### ISSUE TITLE
[Feature Request] Support plugin for loss (in pytorch backend)

### ISSUE THREAD

**Issue by ChiahsinChu** [deepmodeling/deepmd-kit#4189]
### Summary

Currently, the loss function module is chosen by `deepmd.pt.train.training.get_loss` based on the loss type keyword. This means that the user cannot implement a custom loss function module in a plugin, without changing the code in the deepmd-kit package. I wonder whether it is possible to have a hook for loss function module, as what has been done for the descirptors, etc. 

### Detailed Description

See above.
